h4/h4.ino

```diff
@@ -2935,6 +2965,49 @@ bool handleMcode(const String& command) {
   return true;
 }
 
+bool handleChangeToolCommand(String command) {
+  int toolNumber = command.substring(1).toInt();
+  if (toolNumber >= 0 && toolNumber < 4) {
+    changeTool(toolNumber);
+    return true;
+  } else {
+    Serial.println("error: invalid tool number");
+    return false;
+  }
+}
+
+void changeTool(int newToolNumber) {
+    currentTool = newToolNumber;
+    ToolOffset newToolOffset = toolOffsets[newToolNumber];
+
+    // Calculate the net offset needed to apply
+    ToolOffset netOffset = {
+        newToolOffset.xOffset - currentOffset.xOffset,
+        newToolOffset.zOffset - currentOffset.zOffset
+    };
+
+    applyToolOffset(netOffset);
+    currentOffset = newToolOffset;
+
+    Serial.println("Tool changed to T" + String(newToolNumber));
```

Comment:
In GCode we don't confirm like that, instead you should already get `ok` as a reply

Reply:
Agree. This was for debugging forgot to remove.